src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ManageDocumentsCallbackHandlerTest.java

```diff
@@ -96,25 +111,102 @@ void shouldNotReturnErrorsIfThereAreDifferences() {
+                    .type(CallbackType.MID)
+                    .request(request)
+                    .build();
+
+            }
+
+            @Test
+            void shouldReturnErrorsIfThereAreNoDifferencesNullStaffDocuments() {
+
+                CCDCase ccdCase = CCDCase.builder().build();
+
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetails))).thenReturn(ccdCase);
+
+                AboutToStartOrSubmitCallbackResponse response
+                    = (AboutToStartOrSubmitCallbackResponse) handler.handle(callbackParams);
+
+                Assertions.assertNotNull(response.getErrors());
+                assertThat(response.getErrors(), contains(ManageDocumentsCallbackHandler.NO_CHANGES_ERROR_MESSAGE));
+            }
+
+            @Test
+            void shouldReturnErrorsIfThereAreNoDifferencesNullBefore() {
+
+                CCDCase ccdCase = CCDCase.builder()
+                    .build();
+
+                CCDCase ccdCaseBefore = CCDCase.builder()
+                    .staffUploadedDocuments(List.of())
+                    .build();
+
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetails))).thenReturn(ccdCase);
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetailsBefore))).thenReturn(ccdCaseBefore);
+
+                AboutToStartOrSubmitCallbackResponse response
+                    = (AboutToStartOrSubmitCallbackResponse) handler.handle(callbackParams);
+
+                Assertions.assertNotNull(response.getErrors());
+                assertThat(response.getErrors(), contains(ManageDocumentsCallbackHandler.NO_CHANGES_ERROR_MESSAGE));
+            }
+
+            @Test
+            void shouldReturnErrorsIfThereAreNoDifferencesNullAfter() {
+                CCDCase ccdCase = CCDCase.builder()
+                    .staffUploadedDocuments(List.of())
+                    .build();
+
+                CCDCase ccdCaseBefore = CCDCase.builder()
+                    .build();
+
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetails))).thenReturn(ccdCase);
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetailsBefore))).thenReturn(ccdCaseBefore);
+
+                AboutToStartOrSubmitCallbackResponse response
+                    = (AboutToStartOrSubmitCallbackResponse) handler.handle(callbackParams);
+
+                Assertions.assertNotNull(response.getErrors());
+                assertThat(response.getErrors(), contains(ManageDocumentsCallbackHandler.NO_CHANGES_ERROR_MESSAGE));
+            }
+
+            @Test
+            void shouldReturnErrorsIfThereAreNoDifferencesEmptyStaffUploadedDocuments() {
+                CaseDetails caseDetails = CaseDetails.builder().build();
+
+                CCDCase ccdCase = CCDCase.builder()
+                    .staffUploadedDocuments(List.of())
+                    .build();
+
+                when(caseDetailsConverter.extractCCDCase(eq(caseDetails))).thenReturn(ccdCase);
```

Comment:
Would be nice to see mocking of caseDetailsBefore. I guess the assumption is that the value is null.